src/Concordium/Client/Runner.hs

```diff
@@ -2514,226 +3031,215 @@ processBakerCmd action baseCfgDir verbose backend =
             when pubSuccess $ do
               logSuccess [ printf "public keys written to file '%s'" pubFile]
 
-    BakerAdd bakerKeysFile txOpts initialStake autoRestake outputFile -> do
-      baseCfg <- getBaseConfig baseCfgDir verbose
-      when verbose $ do
-        runPrinter $ printBaseConfig baseCfg
-        putStrLn ""
+    BakerAdd bakerKeysFile txOpts initialStake autoRestake openForDelegation metadataURL transactionFeeCommission bakingRewardCommission finalizationRewardCommission outputFile -> do
+      pv <- withClient backend $ do
+        cs <- getConsensusStatus >>= getFromJson
+        return $ Queries.csProtocolVersion cs
+      if pv < Types.P4
+        then
+          processBakerAddCmd baseCfgDir verbose backend txOpts initialStake autoRestake bakerKeysFile outputFile
+        else do
+          let allPresent = case (openForDelegation, metadataURL, transactionFeeCommission, bakingRewardCommission, finalizationRewardCommission) of
+                (Just _, Just _, Just _, Just _, Just _) -> True
+                _ -> False
+          when (not allPresent) $ do
+            logWarn $ ["To add a baker, all of the options\n"
+                 ++ "--open-delegation-for,\n"
+                 ++ "--baker-url,\n"
+                 ++ "--delegation-transaction-fee-commission,\n"
+                 ++ "--delegation-baking-commission,\n"
+                 ++ "--delegation-finalization-commission\nmust be present."]
```

Comment:
It would be great if you printed which ones are missing here.

Something like
```haskell
            logWarn $ ["To add a baker, all options are necessary. The following are missing\n"
                 ++ if isNothing openForDelegation then "--open-delegation-for,\n" else ""
                 ...
```